src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -495,7 +495,7 @@ private Conversion ClassifyStandardImplicitConversion(TypeSymbol source, TypeSym
 
             if (HasBoxingConversion(source, destination, ref useSiteDiagnostics))
             {
-                return Conversion.Boxing;
+                return source.Kind == SymbolKind.TypeParameter ? Conversion.TypeParameterBoxing : Conversion.ValueTypeBoxing;
```

Comment:
Yes.  I believ ethat's the case.  Though another @dotnet/roslyn-compiler  dev should take a look.